src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -253,8 +255,9 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
                 if (Thread.interrupted()) {
                     throw new InterruptedException();
                 }
+                Integer limit = page.getLimit();
```

Comment:
could this be `<= 0` ? in that case we possibly risk unnecessary looping.

Reply:
well if less than 0 bitbucket server is playing silly monkeys